biocrnpyler/chemical_reaction_network.py

```diff
@@ -77,6 +79,31 @@ def add_reactions(self, reactions: Union[Reaction,List[Reaction]], show_warnings
 
             #TODO synchronize Species in the CRN
 
+    @property
+    def initial_condition_dict(self):
+        return self._initial_condition_dict
+
+    @initial_condition_dict.setter
+    def initial_condition_dict(self, init_cond_dict):
+        if init_cond_dict is None:
+            init_cond_dict = {}
+        elif isinstance(init_cond_dict, dict):
+            for s in init_cond_dict:
+                if s not in self.species:
+                    raise ValueError(f"Trying to set the initial condition of a Species {s} not in the CRN")
+
+        self._initial_condition_dict = init_cond_dict
+
+    
+    def update_initial_condition(self, init_cond_dict):
```

Comment:
I'm a bit confused by this, there's already a setter/getter for the initial concentration dict, now we have an `update_initial_condition` function. Why do we need the two?

Reply:
This is a standard issue I have with getters and setters for dictionaries. The setter overwrites the entire dictionary, which is good at instantiation but typically not what we want to occur. The update_initial_condition updates the dictionary without overwriting it, in other words a subset of keys can be added or changed and the other key, value pairs will be untouched.

Reply:
You can implement a setter anyway you want.  I'd just call self._initial_condition_dict.update(new_dict) after basic checks.

Reply:
But then it becomes impossible to completely overwrite the dictionary.

Reply:
Ok, will change it. To reset the dictionary to emtpy you now have to do CRN.initial_concentration_dict = None